pkg/fqdn/namemanager/manager.go

```diff
@@ -109,6 +111,10 @@ func (n *manager) RegisterFQDNSelector(selector api.FQDNSelector) {
 		if metrics.FQDNSelectors.IsEnabled() {
 			metrics.FQDNSelectors.Set(float64(len(n.allSelectors)))
 		}
+		// Required to be asynchronous to avoid a deadlock. The selectorcache calls
+		// namemanager while being locked, and allocating an identity with notifyOwner calls
+		// into the selector cache.
+		go n.allocateIdentity(selector)
```

Comment:
The unboundedness here of launching goroutines can get us into trouble. We have a few options here:

1. Bound via a workerpool of size `runtime.NumCPU()`
2. Bound via a `trigger.Trigger` (single threaded worker that handles identities and is triggered from here)

I'm leaning more towards (2) since it feels like (1) is slightly overkill and (2) is sufficient.